freeipa/src/main/java/com/sequenceiq/freeipa/flow/instance/reboot/action/RebootActions.java

```diff
@@ -0,0 +1,92 @@
+package com.sequenceiq.freeipa.flow.instance.reboot.action;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.api.util.ConverterUtil;
+import com.sequenceiq.cloudbreak.cloud.event.instance.RebootInstancesRequest;
+import com.sequenceiq.cloudbreak.cloud.event.instance.RebootInstancesResult;
+import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.freeipa.converter.cloud.InstanceMetaDataToCloudInstanceConverter;
+import com.sequenceiq.freeipa.flow.instance.InstanceEvent;
+import com.sequenceiq.freeipa.flow.instance.InstanceFailureEvent;
+import com.sequenceiq.freeipa.flow.instance.reboot.RebootContext;
+import com.sequenceiq.freeipa.flow.instance.reboot.RebootEvent;
+import com.sequenceiq.freeipa.flow.instance.reboot.RebootService;
+
+@Configuration
+public class RebootActions {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RebootActions.class);
+
+    @Inject
+    private ConverterUtil converterUtil;
+
+    @Inject
+    private RebootService rebootService;
+
+    @Inject
+    private InstanceMetaDataToCloudInstanceConverter metadataConverter;
+
+    @Bean(name = "REBOOT_STATE")
+    public Action<?, ?> rebootAction() {
+        return new AbstractRebootAction<>(InstanceEvent.class) {
+            @Override
+            protected void doExecute(RebootContext context, InstanceEvent payload, Map<Object, Object> variables) {
+                rebootService.startInstanceReboot(context);
+                sendEvent(context);
+                LOGGER.info("Starting reboot for {}", context.getInstanceMetaData().getInstanceId());
+            }
+
+            @Override
+            protected Selectable createRequest(RebootContext context) {
+                List<CloudInstance> cloudInstances = converterUtil.convertAll(List.of(context.getInstanceMetaData()), CloudInstance.class);
+                return new RebootInstancesRequest<>(context.getCloudContext(), context.getCloudCredential(), cloudInstances);
+            }
+        };
+    }
+
+    @Bean(name = "REBOOT_FINISHED_STATE")
+    public Action<?, ?> rebootFinishedAction() {
+        return new AbstractRebootAction<>(RebootInstancesResult.class) {
+            @Override
+            protected void doExecute(RebootContext context, RebootInstancesResult payload, Map<Object, Object> variables) {
+                rebootService.finishInstanceReboot(context, payload);
+                LOGGER.info("Finished rebooting.");
```

Comment:
please add instanceids to the log, makes investigation easier

Reply:
fixed